sycl/include/sycl/builtins.hpp

```diff
@@ -1377,6 +1377,198 @@ detail::common_rel_ret_t<T> signbit(T x) __NOEXC {
       __sycl_std::__invoke_SignBitSet<detail::internal_rel_ret_t<T>>(x));
 }
 
+// marray relational functions
+template <typename T,
+          typename = detail::enable_if_t<
+              detail::is_mgenfloat<T>::value &&
+                  detail::is_svgenfloat<detail::marray_element_type<T>>::value,
+              T>>
+sycl::marray<bool, detail::marray_size<T>> isequal(T x, T y) __NOEXC {
+  static_assert(x.size() == y.size(), "sycl::marray sizes must be equal.");
```

Comment:
`marray` size is a compile-time known property and it is part of the type. You can't get two different sizes here, no need in that static assertion